components/compliance-service/api/tests/25_nodes_spec.rb

```diff
@@ -597,6 +597,72 @@ def cleanup
     )
     assert_equal(["Donkey Kong", "M$", "test-auto-job-creation"], nodes_list["nodes"])
 
+    # Get the nodes with a department tag without passing the values array:
+    nodes_list = MANAGER_GRPC manager, :search_nodes, Manager::NodeQuery.new(
+      node_manager_id: "e69dc612-7e67-43f2-9b19-256afd385820",
+      query: Manager::Query.new(
+        filter_map: [
+          Common::Filter.new(key: "department"),
+        ]
+      )
+    )
+    assert_equal(["betamaniac", "Donkey Kong"], nodes_list["nodes"])
+
+    # Get the nodes with a department tag:
+    nodes_list = MANAGER_GRPC manager, :search_nodes, Manager::NodeQuery.new(
+      node_manager_id: "e69dc612-7e67-43f2-9b19-256afd385820",
+      query: Manager::Query.new(
+        filter_map: [
+          Common::Filter.new(key: "department", values: [""]),
+        ]
+      )
+    )
+    assert_equal(["betamaniac", "Donkey Kong"], nodes_list["nodes"])
+
+    # Get the nodes without a department tag:
+    nodes_list = MANAGER_GRPC manager, :search_nodes, Manager::NodeQuery.new(
+      node_manager_id: "e69dc612-7e67-43f2-9b19-256afd385820",
+      query: Manager::Query.new(
+        filter_map: [
+          Common::Filter.new(key: "department", values: [""], exclude: true),
+        ]
+      )
+    )
+    assert_equal(["M$", "test-auto-job-creation"], nodes_list["nodes"])
+
+    # Get the nodes tagged market:market% or market:missing-in-action%:
+    nodes_list = MANAGER_GRPC manager, :search_nodes, Manager::NodeQuery.new(
+      node_manager_id: "e69dc612-7e67-43f2-9b19-256afd385820",
+      query: Manager::Query.new(
+        filter_map: [
+          Common::Filter.new(key: "department", values: ["market", "missing-in-action"]),
+        ]
+      )
+    )
```

Comment:
Is there a missing assert here?

Reply:
Yes, good catch. Pushed another commit with the assert and also updated the comments for the two department tag filters.